lib/descriptors/FieldDescriptor.js

```diff
@@ -131,6 +137,22 @@ FieldDescriptor.prototype.isRepeated = function () {
 }
 
 
+FieldDescriptor.prototype.setOneofIndex = function (oneofIndex) {
+  this._oneofIndex = oneofIndex
```

Comment:
oh i guess this is what protoc does
https://github.com/google/protobuf/blob/master/src/google/protobuf/descriptor.proto#L191
carry on!